Unit test failure on master: 1b91a780e1bbc9da6daf8ac989dc2f48bafa24e1

PASS
coverage: 46.0% of statements
ok  	github.com/vmware/vic/cmd/imagec	0.035s
+ for pkg in '"$@"'
++ echo github.com/vmware/vic/cmd/vicadmin
++ tr / -
+ f=/drone/src/github.com/vmware/vic/.cover/github.com-vmware-vic-cmd-vicadmin.cover
+ go test -v -covermode=count -coverprofile=/drone/src/github.com/vmware/vic/.cover/github.com-vmware-vic-cmd-vicadmin.cover github.com/vmware/vic/cmd/vicadmin
time="2016-06-30T23:18:53Z" level=debug msg="[BEGIN] [github.com/vmware/vic/cmd/vicadmin.init.1]" 
time="2016-06-30T23:18:53Z" level=error msg="Unable to load configuration from guestinfo" 
time="2016-06-30T23:18:53Z" level=debug msg="[ END ] [github.com/vmware/vic/cmd/vicadmin.init.1] [240.324µs] " 
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0xa9a9f2]

goroutine 1 [running]:
panic(0x1014d00, 0xc820012180)
	/usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/vmware/vic/pkg/vsphere/extraconfig.decodeString(0x0, 0xd28f20, 0x1d885e0, 0x198, 0xc8201da840, 0x2b, 0xfffffffffffffffa, 0x1, 0x0, 0x0, ...)
	/drone/src/github.com/vmware/vic/pkg/vsphere/extraconfig/decode.go:88 +0x62
github.com/vmware/vic/pkg/vsphere/extraconfig.decode(0x0, 0xd28f20, 0x1d885e0, 0x198, 0xc8201da840, 0x2b, 0xfffffffffffffffa, 0x1, 0x0, 0x0, ...)
	/drone/src/github.com/vmware/vic/pkg/vsphere/extraconfig/decode.go:78 +0x253
github.com/vmware/vic/pkg/vsphere/extraconfig.decodeStruct(0x0, 0x11895a0, 0x1d885e0, 0x199, 0xc8200bdd40, 0x16, 0xfffffffffffffffb, 0x1, 0x0, 0x0, ...)
	/drone/src/github.com/vmware/vic/pkg/vsphere/extraconfig/decode.go:196 +0x8d7
github.com/vmware/vic/pkg/vsphere/extraconfig.decode(0x0, 0x11895a0, 0x1d885e0, 0x199, 0xc8200bdd40, 0x16, 0xfffffffffffffffb, 0x1, 0x0, 0x0, ...)
	/drone/src/github.com/vmware/vic/pkg/vsphere/extraconfig/decode.go:78 +0x253
github.com/vmware/vic/pkg/vsphere/extraconfig.decodeStruct(0x0, 0x11b2640, 0x1d885e0, 0x199, 0xc820197150, 0xf, 0xfffffffffffffffc, 0x1, 0x0, 0x0, ...)
	/drone/src/github.com/vmware/vic/pkg/vsphere/extraconfig/decode.go:196 +0x8d7
github.com/vmware/vic/pkg/vsphere/extraconfig.decode(0x0, 0x11b2640, 0x1d885e0, 0x199, 0xc820197150, 0xf, 0xfffffffffffffffc, 0x1, 0x0, 0x0, ...)
	/drone/src/github.com/vmware/vic/pkg/vsphere/extraconfig/decode.go:78 +0x253
github.com/vmware/vic/pkg/vsphere/extraconfig.decodeStruct(0x0, 0x1233d20, 0x1d885e0, 0x199, 0x0, 0x0, 0xfffffffffffffffd, 0x49c801, 0x0, 0x0, ...)
	/drone/src/github.com/vmware/vic/pkg/vsphere/extraconfig/decode.go:196 +0x8d7
github.com/vmware/vic/pkg/vsphere/extraconfig.decode(0x0, 0x1233d20, 0x1d885e0, 0x199, 0x0, 0x0, 0xfffffffffffffffd, 0x455c01, 0x0, 0x0, ...)
	/drone/src/github.com/vmware/vic/pkg/vsphere/extraconfig/decode.go:78 +0x253
github.com/vmware/vic/pkg/vsphere/extraconfig.decodePtr(0x0, 0x120a3e0, 0x1d885e0, 0x16, 0x0, 0x0, 0xfffffffffffffffe, 0x40f301, 0x0, 0x0, ...)
	/drone/src/github.com/vmware/vic/pkg/vsphere/extraconfig/decode.go:141 +0x2ab
github.com/vmware/vic/pkg/vsphere/extraconfig.decode(0x0, 0x120a3e0, 0x1d885e0, 0x16, 0x0, 0x0, 0xfffffffffffffffe, 0x7f8000000001, 0x0, 0x0, ...)
	/drone/src/github.com/vmware/vic/pkg/vsphere/extraconfig/decode.go:78 +0x253
github.com/vmware/vic/pkg/vsphere/extraconfig.Decode(0x0, 0x120a3e0, 0x1d885e0, 0x0, 0x0)
	/drone/src/github.com/vmware/vic/pkg/vsphere/extraconfig/decode.go:440 +0xf6
github.com/vmware/vic/cmd/vicadmin.init.1()
	github.com/vmware/vic/cmd/vicadmin/_test/_obj_test/vicadm.go:119 +0x5ad
github.com/vmware/vic/cmd/vicadmin.init()
	/drone/src/github.com/vmware/vic/cmd/vicadmin/vicadm_test.go:276 +0xfd
main.init()
	github.com/vmware/vic/cmd/vicadmin/_test/_testmain.go:108 +0x56
exit status 2
FAIL	github.com/vmware/vic/cmd/vicadmin	0.014s
Makefile:205: recipe for target 'test' failed
make: *** [test] Error 1
[info] build failed (exit code 2)

--- a/cmd/vicadmin/vicadm.go
+++ b/cmd/vicadmin/vicadm.go
@@ -115,7 +115,9 @@ func init() {
 	src, err := extraconfig.GuestInfoSource()
 	if err != nil {
 		log.Errorf("Unable to load configuration from guestinfo")
+		return
 	}
+
 	extraconfig.Decode(src, &vchConfig)
 }
 

--- a/pkg/vsphere/extraconfig/decode.go
+++ b/pkg/vsphere/extraconfig/decode.go
@@ -435,6 +435,12 @@ type DataSource func(string) (string, error)
 // Decode populates a destination with data from the supplied data source
 func Decode(src DataSource, dest interface{}) interface{} {
 	defer log.SetLevel(log.GetLevel())
+
+	if src == nil {
+		log.Warnf("Decode source is nil - unable to continue")
+		return dest
+	}
+
 	log.SetLevel(DecodeLogLevel)
 
 	value := decode(src, reflect.ValueOf(dest), DefaultPrefix, Unbounded)
@@ -446,6 +452,12 @@ func Decode(src DataSource, dest interface{}) interface{} {
 // the specified prefix - this allows for decode into substructres.
 func DecodeWithPrefix(src DataSource, dest interface{}, prefix string) interface{} {
 	defer log.SetLevel(log.GetLevel())
+
+	if src == nil {
+		log.Warnf("Decode source is nil - unable to continue")
+		return dest
+	}
+
 	log.SetLevel(DecodeLogLevel)
 
 	value := decode(src, reflect.ValueOf(dest), prefix, Unbounded)

